compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.cpp

```diff
@@ -313,9 +316,37 @@ std::string stringifyOperandIndex(IntegerAttr valueAttr) {
 Value PadEncodingLayoutAttr::calculateStorageSizeInBytes(
     Location loc, OpBuilder &builder, RankedTensorType type,
     ValueRange dynamicDims) const {
-  // TODO(kuhar): Add sizeof calculation.
-  assert(false && "Unimplemented");
-  return nullptr;
+  ArrayRef<int32_t> padding = getPadding().asArrayRef();
+  assert(padding.size() == type.getRank() && "Invalid padding");
+
+  const int64_t elementSize =
+      llvm::divideCeil(type.getElementTypeBitWidth(), 8);
+  int64_t staticProduct = elementSize;
+  Value dynamicProduct = builder.create<arith::ConstantIndexOp>(loc, 1);
+
+  size_t dynamicDimIdx = 0;
+  for (auto [dimSize, padValue] : llvm::zip_equal(type.getShape(), padding)) {
+    if (!ShapedType::isDynamic(dimSize)) {
+      staticProduct *= (dimSize + padValue);
+      continue;
+    }
+
+    Value dynamicDimSize = dynamicDims[dynamicDimIdx];
+    ++dynamicDimIdx;
+
+    if (padValue != 0) {
+      dynamicDimSize = builder.create<arith::AddIOp>(
+          loc, dynamicDimSize,
+          builder.create<arith::ConstantIndexOp>(loc, padValue),
+          arith::IntegerOverflowFlags::nsw);
```

Comment:
This is new to me. Does it add overheads to the computation?

Reply:
No, it allows us to assume that none of this indexing map overflows, which should allow for more folds. Not strictly necessary, but we started adding these flags to index computation across the codebase. cc: @MaheshRavishankar @krzysz00

Reply:
I see, noted!